spec/unit/puppet-lint/plugins/legacy_facts/legacy_facts_spec.rb

```diff
@@ -165,6 +173,23 @@
       end
     end
 
+    context "fact variable using legacy $::facts['osfamily']" do
+      let(:code) { "$::facts['osfamily']" }
+      let(:msg) { "legacy fact 'osfamily'" }
+
+      it 'only detects a single problem' do
+        expect(problems).to have(1).problem
+      end
+
+      it 'fixes the problem' do
+        expect(problems).to contain_fixed(msg).on_line(1).in_column(1)
+      end
+
+      it 'uses the facts hash' do
+        expect(manifest).to eq("$facts['os']['family']")
```

Comment:
in the future we should consider reworking the legacy_fact/topscope_fact/topscope_variable check because they heavily overlap.

Reply:
would it make sense to add some tests from https://github.com/puppetlabs/puppet-lint/blob/e3815cebdfa50100acb888e29fde8f03f4e1879b/spec/unit/puppet-lint/plugins/top_scope_facts/top_scope_facts_spec.rb here? But expect no changes? I want to prevent that legacy_fact and topscope_fact fix the same code.

Reply:
I was looking in to this yesterday. It would seem that they no longer clash but adding extra tests is never a bad thing.